core/chains/orm.go

```diff
@@ -164,10 +202,10 @@ func NewNodesORM[ID, NEW, N any](q pg.Q, prefix string, nodeCols ...string) *Nod
 		VALUES (:name, :%s_chain_id, %s, now(), now())
 		RETURNING *;`, prefix, prefix, strings.Join(nodeCols, ", "), prefix, strings.Join(withColon, ", "))
 
-	return &NodesORM[ID, NEW, N]{q: q, prefix: prefix, createNodeQ: query}
+	return &nodesORM[I, N]{q: q, prefix: prefix, createNodeQ: query}
 }
 
-func (o *NodesORM[ID, NEW, N]) CreateNode(data NEW, qopts ...pg.QOpt) (node N, err error) {
+func (o *nodesORM[I, N]) CreateNode(data N, qopts ...pg.QOpt) (node N, err error) {
```

Comment:
The separate `NEW` type doesn't add much value at this level, since the query uses names mapped to fields already anyways, so simpler to reuse `N`.